src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -302,6 +311,21 @@ public class OpenIdConnectAuthenticationHandler : AuthenticationHandler<OpenIdCo
                     _configuration = await Options.ConfigurationManager.GetConfigurationAsync(Context.RequestAborted);
                 }
 
+                // Redeeming authorization code for tokens
+                if (string.IsNullOrEmpty(message.IdToken) && !string.IsNullOrEmpty(message.Code))
```

Comment:
Hum, this check is odd. Why not retrieving an access/identity token from the token endpoint when an id_token has been received from the authorization endpoint? (`string.IsNullOrEmpty(message.IdToken)`)